tests/dimensions.cpp

```diff
@@ -0,0 +1,383 @@
+                alpaka::vec::Vec<Dim, Idx>{Idx{1}, Idx{1}, Idx{1}}},
+            [] ALPAKA_FN_ACC(
+                const Acc & acc,
+                int N,
+                typename ScatterAllocator::AllocatorHandle allocHandle) {
+                const auto idx
+                    = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+                deviceArray[idx[0] * N * N * N * N + idx[1] * N * N + idx[0]]
+                    = (int *)allocHandle.malloc(acc, sizeof(int));
+            },
+            N,
+            scatterAlloc.getAllocatorHandle()));
+
+    const auto slots = scatterAlloc.getAvailableSlots(dev, queue, sizeof(int));
+    const auto heapInfo = scatterAlloc.getHeapLocations().at(0);
+    std::cout << alpaka::acc::traits::GetAccName<Acc>::getAccName()
+              << " slots: " << slots << " heap size: " << heapInfo.size << '\n';
+
+    // free N*N*N * N*N*N allocations from N*N*N blocks of N*N*N threads for
+    // ints
+    alpaka::queue::enqueue(
+        queue,
+        alpaka::kernel::createTaskKernel<Acc>(
+            alpaka::workdiv::WorkDivMembers<Dim, Idx>{
+                alpaka::vec::Vec<Dim, Idx>{Idx{N}, Idx{N}, Idx{N}},
+                alpaka::vec::Vec<Dim, Idx>{Idx{N}, Idx{N}, Idx{N}},
+                alpaka::vec::Vec<Dim, Idx>{Idx{1}, Idx{1}, Idx{1}}},
+            [] ALPAKA_FN_ACC(
+                const Acc & acc,
+                int N,
+                typename ScatterAllocator::AllocatorHandle allocHandle) {
+                const auto idx
+                    = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+                allocHandle.free(
+                    acc,
+                    deviceArray
+                        [idx[0] * N * N * N * N + idx[1] * N * N + idx[0]]);
+            },
+            N,
+            scatterAlloc.getAllocatorHandle()));
+
+    // free 1 allocation from 1 thread for N*N*N * N*N*N pointers
+    alpaka::queue::enqueue(
+        queue,
+        alpaka::kernel::createTaskKernel<Acc>(
+            alpaka::workdiv::WorkDivMembers<Dim, Idx>{
+                alpaka::vec::Vec<Dim, Idx>{Idx{1}, Idx{1}, Idx{1}},
+                alpaka::vec::Vec<Dim, Idx>{Idx{1}, Idx{1}, Idx{1}},
+                alpaka::vec::Vec<Dim, Idx>{Idx{1}, Idx{1}, Idx{1}}},
+            [] ALPAKA_FN_ACC(
+                const Acc & acc,
+                typename ScatterAllocator::AllocatorHandle allocHandle) {
+                allocHandle.free(acc, deviceArray);
+            },
+            scatterAlloc.getAllocatorHandle()));
+}
+
+auto main(int argc, char ** argv) -> int
+try
+{
+    test1D<alpaka::acc::AccGpuCudaRt>();
+    test2D<alpaka::acc::AccGpuCudaRt>();
+    test3D<alpaka::acc::AccGpuCudaRt>();
+
+    test1D<alpaka::acc::AccCpuThreads>();
+    test2D<alpaka::acc::AccCpuThreads>();
+    test3D<alpaka::acc::AccCpuThreads>();
+
+    test1D<alpaka::acc::AccCpuOmp2Threads>();
+    test2D<alpaka::acc::AccCpuOmp2Threads>();
+    test3D<alpaka::acc::AccCpuOmp2Threads>();
```

Comment:
I would like to have a unit test framework for running these. I saw alpaka uses Catch2. Would you and how would you integrate it here as well?
If you are not certain about Catch2, I am also a big fan of gtest.

Reply:
Yes, alpaka uses Catch2 and I think the idea was that other projects that did not commit to any framework yet should (but not have to) also prefer this. What do you think @psychocoderHPC ?

Reply:
I adopted Catch2 and split main() into several test cases.